src/charm.py

```diff
@@ -301,18 +305,70 @@ def _prometheus_config(self):
 
         return yaml.dump(scrape_config)
 
+    def _get_image_tag(self, api_tags=None) -> str:
+        """Get most recent patch of major and minor Prometheus version
+        """
+        # 1. get valid tags from URL (or directly from input)
+        # 2. clean tags data structure and filter out release candidates "rc"
+        # 3. filter out tags that don't match our major/minor specification
+        config = self.model.config
+        if not api_tags:
+            api_tags_resp = requests.get(IMAGE_TAGS_URL)
+            if not api_tags_resp.ok:
+                logger.error('No Prometheus image tag can be selected. Cannot access '
+                             '{} (defined in config.yaml).'.format(IMAGE_TAGS_URL))
+                self.unit.status = BlockedStatus('Cannot get container image path')
+                return ''
+            api_tags = json.loads(api_tags_resp.text)
+
+        try:
+            major, minor = config['prometheus-image-version'].split('.')
+        except ValueError:
+            logger.error('Prometheus version must be in the form {major}.{minor}')
+            self.unit.status = BlockedStatus('Incorrect Prometheus version format')
+            return ''
+
+        # semver doesn't like 'v' at the beginning of the version string
+        # so we'll create a map of invalid semver strings to valid semver strings
+        tags_map = {tag['name']: tag['name'][1::] if tag['name'][0] == 'v' else tag['name']
+                    for tag in api_tags}
+
+        # get list of tuples of the structure [ (original_tag, parsed_version_object), ... ]
```

Comment:
I save the original tag because that's what is ultimately returned from the function and parsed_version is used for sorting.

Reply:
I see the problem now. The problem is the inconsistent use of the "v" prefix by Prometheus docker images. If they hand not done this it would have been easy to just reassemble the original version tag from the parsed version tag.